src/main/java/com/google/cloud/spanner/pgadapter/ProxyServer.java

```diff
@@ -91,8 +91,12 @@ public void startServer() {
           String.valueOf(this.options.getProxyPort()));
       this.status = ServerStatus.STARTING;
       start();
-      logger.log(
-          Level.INFO, "Server started on port {0}", String.valueOf(this.options.getProxyPort()));
+      // TODO: Consider changing the server to use ApiService, so we don't have to manage the state
+      //  of the server manually.
+      while (this.status == ServerStatus.STARTING) {
```

Comment:
We should wait until the server has actually started before logging the port that is being used, and before returning so the caller is not trying to connect to the server before it has actually started.